src/ray/common/client_connection.cc

```diff
@@ -28,14 +28,24 @@ ServerConnection<T>::ServerConnection(boost::asio::basic_stream_socket<T> &&sock
 template <class T>
 void ServerConnection<T>::WriteBuffer(
     const std::vector<boost::asio::const_buffer> &buffer, boost::system::error_code &ec) {
-  boost::asio::write(socket_, buffer, ec);
+  // Loop until all bytes are written, and handle system errors.
```

Comment:
Great find.  Can you also add a longer comment here to say something like, "When profiling with pprof, unhandled interrupts would ..."?  Usually a good idea to add sufficient details to remind others/self for tricky issues like this one ;)